game/src/Player/PassThrough.gd

```diff
@@ -0,0 +1,19 @@
+extends Area2D
+
+const PASS_THROUGH_LAYER = 3
+
+func _ready() -> void:
+	connect("body_exited", self, "_on_body_exited")
+
+
+func _on_body_exited(body: CollisionObject2D) -> void:
+	if not owner.get_collision_mask_bit(PASS_THROUGH_LAYER):
+		owner.set_collision_mask_bit(PASS_THROUGH_LAYER, true)
+
+
+func _unhandled_input(event: InputEvent) -> void:
+	if event.is_action_pressed("move_down") and owner.is_on_floor():
+		owner.set_collision_mask_bit(PASS_THROUGH_LAYER, false)
+		owner.state_machine.transition_to("Move/Air")
```

Comment:
If it's a detector (or a toggle in this case), it shouldn't change the state of the player. Nothing other than states should trigger transitions to other states. That's why you need something to happen in idle or run, or their parent, Move.